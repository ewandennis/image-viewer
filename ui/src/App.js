import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import CameraIcon from '@material-ui/icons/PhotoCamera';
import CssBaseline from '@material-ui/core/CssBaseline';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import ImageGrid from './ImageGrid';

const useStyles = makeStyles(theme => ({
  toolbarIcon: {
    marginRight: theme.spacing(2)
  },
  main: {
    paddingTop: theme.spacing(6)
  }
}));

function App() {
  const classes = useStyles();
  return (
    <>
      <CssBaseline />
      <AppBar position="relative">
        <Toolbar>
          <CameraIcon className={classes.toolbarIcon} />
          <Typography variant="h4">Image Viewer</Typography>
        </Toolbar>
      </AppBar>
      <main className={classes.main}>
        <ImageGrid />
      </main>
    </>
  );
}

export default App;
