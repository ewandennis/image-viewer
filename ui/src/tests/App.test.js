import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

jest.mock('socket.io-client');

describe('Image Viewer', () => {
  it('renders', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
