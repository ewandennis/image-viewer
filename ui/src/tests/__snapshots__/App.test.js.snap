// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Viewer renders 1`] = `
<Fragment>
  <CssBaseline />
  <WithStyles(ForwardRef(AppBar))
    position="relative"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <PhotoCameraIcon
        className="makeStyles-toolbarIcon-1"
      />
      <WithStyles(ForwardRef(Typography))
        variant="h4"
      >
        Image Viewer
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <main
    className="makeStyles-main-2"
  >
    <ImageGrid />
  </main>
</Fragment>
`;
